omod/src/main/webapp/pages/allergy.gsp

```diff
@@ -73,18 +73,18 @@ ${ ui.includeFragment("allergyui", "removeAllergyDialog") }
                         <% allergens.each { allergen -> %>
                         <li>
                             <% if (allergen.id == otherNonCodedConcept.id) { %>
-                                <input id="allergen-${allergen.id}" type="radio" name="codedAllergen" value="${allergen.id}" class="coded_allergens" ng-model="allergen" ng-click="checkOtherRadioButton()"
+                             <input id="allergen-${ typeName }" type="radio" name="codedAllergen" value="${allergen.id}" ng-click="selectOtherAllergenRadioInput()" class="coded_allergens" ng-model="allergen"
```

Comment:
Well done @Muta-Jonathan , why removing allergen.id to typeName.

Reply:
I tried @sherrif10 but when I used allergen.id the radio button doesn't show... but otherwise let me look in it more